src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -133,23 +133,24 @@ private static IntPtr HostFxrResolver(Assembly assembly, string libraryName)
             var hostFxrLibName = "libhostfxr";
             var libExtention = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "dylib" : "so";
 
-            if (!hostFxrLibName.Equals(libraryName, StringComparison.OrdinalIgnoreCase))
+            if (!hostFxrLibName.Equals(libraryName))
             {
                 return IntPtr.Zero;
             }
 
             var hostFxrRoot = Path.Combine(DotnetPath.Value, "host", "fxr");
             if (Directory.Exists(hostFxrRoot))
             {
-                // Agreed to load hostfxr from the highest version
+                var versionParser = new SemanticVersionParser();
+                // Load hostfxr from the highest version, because it should be backward-compatible
                 var hostFxrAssemblyDirectory = Directory.GetDirectories(hostFxrRoot)
-                            .Select(str => NuGetVersion.Parse(str))
+                            .Select(str => versionParser.TryParse(str, out var version) ? version : null)
                             .Max();
```

Comment:
Nit: if you used `.OrderBy().First()` you would have a `DirectoryInfo` you could call `GetFiles` on which should be slightly more efficient.